### PR TITLE
add "to have size" assertion

### PR DESCRIPTION
**Pull request by boneskull**
pretty straightforward.  let me know if there's a different canonical way to implement

### CODE REVIEW

File: lib/unexpectedMap.js [hunk 1]
```diff
@@ -505,6 +505,14 @@ function createUnexpectedMap(expect) {
       );
     }
   );
+
+  expect.addAssertion('<Map> [not] to have size <number>', (expect, subject, expectedSize) => {
+    if (expect.flags.not) {
+      expect(subject.size === expectedSize, 'to be false');
+    } else {
+      expect(subject.size === expectedSize, 'to be true');  
+    }
```
Comment (papandreou) [inline lib/unexpectedMap.js]: If you use `to equal` there'll be a diff that says eg. `expected 1 to equal 4`, which might make it easier to understand why the expectation fails when the set has many members.

Also, you can use [flag forwarding](https://unexpected.js.org/api/addAssertion/#flags) to simplify it a bit:

```suggestion
    expect(subject.size, '[not] to equal', expectedSize);
```

**Comment by papandreou** [comment]
Looks good! Just missing docs :)

**Comment by alexjeffburke** [comment]
> Looks good! Just missing docs :)

@papandreou this is the one plug-in repo that does not currently have docs. I think I've got w branch somewhere with some bits - but let's merge this as-is and I can add something basic later if need be.

**Comment by papandreou** [comment]
Ah, okay :)

**Review by papandreou — APPROVED** [review]
LGTM :rocket:

**Comment by boneskull** [comment]
yah, I was gonna say... "what docs?" 😄

**Comment by alexjeffburke** [comment]
Released as version 2.1.0.

**Pull request merged**